### PR TITLE
Fix _size problem for latest Django

### PR DESCRIPTION
**Pull request by lqez**
`_size` should not be used for getting the file size. Or we get `AttributeError: 'InMemoryUploadedFile' object has no attribute '_size'` error with latest Django.
XRef: https://github.com/django/django/pull/9769/files

### CODE REVIEW

**Comment by timgraham** [comment]
Why use `len()` instead of `.size`? Can you add regression tests?

**Comment by lqez** [comment]
The build error occurred on Docker/Travis side.
https://travis-ci.org/neutronX/django-markdownx/jobs/434021654

File: markdownx/tests/tests.py [hunk 1]
```diff
@@ -1,8 +1,35 @@
...
+
+            try:
+                # Django >= 1.9
+                json = response.json()
+            except AttributeError:
+                import json
+                json = json.loads(response.content.decode('utf-8'))
+
+            self.assertEqual(response.status_code, 200)
+            self.assertTrue('image_code' in json)
```
Comment (timgraham) [inline markdownx/tests/tests.py]: assertIn

File: markdownx/tests/tests.py [hunk 2]
```diff
@@ -1,8 +1,35 @@
 from django.test import TestCase
+try:
+    # Django >= 2.0
+    from django.urls import reverse
+except ImportError:
+    from django.core.urlresolvers import reverse
+import os
```
Comment (timgraham) [inline markdownx/tests/tests.py]: Sort stdlib imports above the Django imports.

File: markdownx/tests/tests.py [hunk 3]
```diff
@@ -1,8 +1,35 @@
 from django.test import TestCase
+try:
+    # Django >= 2.0
+    from django.urls import reverse
+except ImportError:
```
Comment (timgraham) [inline markdownx/tests/tests.py]: I usually use this style: `except ImportError:  # Djanago <  2.0`

File: markdownx/tests/tests.py [hunk 4]
```diff
@@ -1,8 +1,35 @@
...
 
 class SimpleTest(TestCase):
 
     def test_me(self):
         response = self.client.get('/testview/')
         self.assertEqual(response.status_code, 200)
+
+    def test_upload(self):
+        url = reverse('markdownx_upload')
+
```
Comment (timgraham) [inline markdownx/tests/tests.py]: I'd omit the blank lines.

File: markdownx/tests/tests.py [hunk 5]
```diff
@@ -1,8 +1,35 @@
...
+        with open('markdownx/tests/static/django-markdownx-preview.png', 'rb') as fp:
+            response = self.client.post(url, {'image': fp}, HTTP_X_REQUESTED_WITH='XMLHttpRequest')
+
+            try:
```
Comment (timgraham) [inline markdownx/tests/tests.py]: I think the rest of the test can be deindented. It doesn't need to be in the "with" context manager.

File: markdownx/tests/tests.py [hunk 6]
```diff
@@ -1,8 +1,35 @@
...
+                # Django >= 1.9
+                json = response.json()
+            except AttributeError:
+                import json
+                json = json.loads(response.content.decode('utf-8'))
+
+            self.assertEqual(response.status_code, 200)
+            self.assertTrue('image_code' in json)
+
+            match = re.findall(r'^\!\[\]\(([\w\\\/\-\.]+)\)', json['image_code'])
```
Comment (timgraham) [inline markdownx/tests/tests.py]: Maybe there's a better way to do this. The regex looks a bit too complicated. Also, it should be in try/finally or something so cleanup happens even if the test exits early.

Is `markdownx/15a2f283-a906-4d30-a461-61e1d6abfa20.png` meant to be added in this PR?

**Comment by lqez** [comment]
@timgraham Thanks for your review. I updated PR.

**Pull request merged**